x-pack/plugins/infra/server/routes/custom_dashboards/get_custom_dashboard.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createRouteValidationFunction } from '@kbn/io-ts-utils';
+import {
+  InfraGetCustomDashboardsRequestParamsRT,
+  InfraGetCustomDashboardsResponseBodyRT,
+} from '../../../common/http_api/custom_dashboards_api';
+import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { handleRouteErrors } from '../../utils/handle_route_errors';
+import { checkCustomDashboardsEnabled } from './lib/check_custom_dashboards_enabled';
+import { findCustomDashboard } from './lib/find_custom_dashboard';
+
+export function initGetCustomDashboardRoute(framework: KibanaFramework) {
+  const validateParams = createRouteValidationFunction(InfraGetCustomDashboardsRequestParamsRT);
+
+  framework.registerRoute(
+    {
+      method: 'get',
+      path: '/api/infra/custom-dashboards/{assetType}',
+      validate: {
+        params: validateParams,
+      },
+    },
```

Comment:
I'm not familiar with this framework adapter; is this endpoint intended to be `internal`? By default it will be on serverless, but I think it may be worth making it `access: 'internal'` intentionally if we don't expect third part integration.

Reply:
Yeah, good point, I'll see how to make it internal 🙌